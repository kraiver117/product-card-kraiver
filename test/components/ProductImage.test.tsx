import React from 'react';
import renderer from 'react-test-renderer';
import { ProductImage } from '../../src/components';
import ProductCard from '../../src/components/index';
import { product2 } from '../data/products';

describe('Product Image', () => {
  test('should render an product image with image', () => {
    const wrapper = renderer.create(
      <ProductImage img={product2.img} />
    )

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('should render a product card with image', () => {
    const wrapper = renderer.create(
      <ProductCard product={product2}>
        {
          ({ product }) => (
            <>
              <ProductImage img={product.img} />
            </>
          )
        }
      </ProductCard>
    )

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
