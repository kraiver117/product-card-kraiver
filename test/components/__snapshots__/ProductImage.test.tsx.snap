// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Image should render a product card with image 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product"
    className="productImg undefined"
    src="./coffe-mug.png"
  />
</div>
`;

exports[`Product Image should render an product image with image 1`] = `
<img
  alt="Product"
  className="productImg undefined"
  src="./coffe-mug.png"
/>
`;
